### PR TITLE
taskfile.dev: Add page with API Reference

### PR DESCRIPTION
**Pull request by andreynering**
This PR adds a new page to the documentation with CLI and schema API reference.

The idea was from @abrahammurciano, and it makes total sense.

This will only be published to https://taskfile.dev once this reaches the master branch. For now, you can preview this page here: https://preview-taskfile.netlify.app/api/

Let me know if you have any additional suggestions.

### CODE REVIEW

**Comment by abrahammurciano** [comment]
Wonderful! This will be so helpful thanks so much!

**Review by ghostsquad — APPROVED** [review]

File: docs/docs/api_reference.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+|      | `--output-group-begin` | `string` | | Message template to print before a task's grouped output. |
+|      | `--output-group-end `  | `string` | | Message template to print after a task's grouped output. |
+| `-p` | `--parallel` | `bool` | | Executes tasks provided on command line in parallel. |
+| `-s` | `--silent` | `bool` | | Disables echoing. |
+|      | `--status` | `bool` | | Exits with non-zero exit code if any of the given tasks is not up-to-date. |
+|      | `--summary` | `bool` | | Show summary about a task. |
+| `-t` | `--taskfile` | `string` | | Choose which Taskfile to run. Defaults to `Taskfile.yml`. |
+| `-v` | `--verbose` | `bool` | | Enables verbose mode. |
+|      | `--version` | `bool` | | Show Task version. |
+| `-w` | `--watch` | `bool` | | Enables watch of the given task. |
```
Comment (ghostsquad) [inline docs/docs/api_reference.md:44]: Most of these options have defaults.

File: docs/docs/api_reference.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+| Attribute | Type | Default | Description |
+| - | - | - | - |
+| `version` | `string` | | Version of the Taskfile. The current version is `3`. |
+| `includes` | [`map[string]Include`](#include) | | Additional Taskfiles to be included. |
+| `output` | `string` | `interleaved` | Output mode. Available options: `interleaved`, `group` and `prefixed`. |
+| `method` | `string` | `checksum` | Default method in this Taskfile. Can be overriden in a task by task basis. Available options: `checksum`, `timestamp` and `none`. |
+| `silent` | `bool` | `false` | Default "silent" options for this Taskfile. If `false`, can be overidden with `true` in a task by task basis. |
+| `run` | `string` | `always` | Default "run" option for this Taskfile. Available options: `always`, `once` and `when_changed`. |
+| `vars` | [`map[string]Variable`](#variable) | | Global variables. |
```
Comment (ghostsquad) [inline docs/docs/api_reference.md:58]: should the default here be either `nil` or `{}` ?

File: docs/docs/api_reference.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+| `dotenv` | `[]string` | | A list of `.env` file paths to be parsed. |
+| `tasks` | [`map[string]Task`](#task) | | The task definitions. |
+
+
+### Include
+
+| Attribute | Type | Default | Description |
+| - | - | - | - |
+| `taskfile` | `string` | | The path for the Taskfile or directory to be included. If a directory, Task will look for files named `Taskfile.yml` or `Taskfile.yaml` inside that directory. |
+| `dir` | `string` | | The working directory of the included tasks when run. Defaults to the parent Taskfile directory. |
```
Comment (ghostsquad) [inline docs/docs/api_reference.md:69]: Should `Defaults to the parent Taskfile directory` be rewritten as `The parent Taskfile directory` and moved into the `Default` column?

**Review by ghostsquad — DISMISSED** [review]

**Review by ghostsquad — COMMENTED** [review]
A few comments about defaults, otherwise looks good.

File: docs/docs/api_reference.md (hunk 2 above)
Comment (andreynering) [reply docs/docs/api_reference.md:58]: I think we probably don't need to specify when the value would be empty.

**Pull request merged**